types/artifacts.d.ts

```diff
@@ -5,892 +5,886 @@
-      /** Source maps of scripts executed in the page. */
-      SourceMaps: Array<Artifacts.SourceMap>;
-      /** Information on <script> and <link> tags blocking first paint. */
-      TagsBlockingFirstPaint: Artifacts.TagBlockingFirstPaint[];
-      /** Information about tap targets including their position and size. */
-      TapTargets: Artifacts.TapTarget[];
-      /** The primary log of devtools protocol activity. Used in Fraggle Rock gathering. */
-      Trace: LH.Trace;
-      /** Elements associated with metrics (ie: Largest Contentful Paint element). */
-      TraceElements: Artifacts.TraceElement[];
-    }
+/**
+ * The set of base artifacts that are available in every mode of Lighthouse operation.
+ */
+interface UniversalBaseArtifacts {
+  /** The ISO-8601 timestamp of when the test page was fetched and artifacts collected. */
+  fetchTime: string;
+  /** A set of warnings about unexpected things encountered while loading and testing the page. */
+  LighthouseRunWarnings: Array<string | IcuMessage>;
+  /** The benchmark index that indicates rough device class. */
+  BenchmarkIndex: number;
+  /** An object containing information about the testing configuration used by Lighthouse. */
+  settings: Config.Settings;
+  /** The timing instrumentation of the gather portion of a run. */
+  Timing: Artifacts.MeasureEntry[];
+}
 
-    export type ArbitraryEqualityMap = _ArbitraryEqualityMap;
+/**
+ * The set of base artifacts whose semantics differ or may be valueless in certain Lighthouse gather modes.
+ */
+interface ContextualBaseArtifacts {
+  /** The URL initially requested and the post-redirects URL that was actually loaded. */
+  URL: {requestedUrl: string, finalUrl: string};
+  /** If loading the page failed, value is the error that caused it. Otherwise null. */
+  PageLoadError: LighthouseError | null;
+}
 
-    module Artifacts {
-      export type ComputedContext = Immutable<{
-        computedCache: Map<string, ArbitraryEqualityMap>;
-      }>;
+/**
+ * The set of base artifacts that were replaced by standard gatherers in Fraggle Rock.
+ */
+interface LegacyBaseArtifacts {
+  /** Device which Chrome is running on. */
+  HostFormFactor: 'desktop'|'mobile';
+  /** The user agent string of the version of Chrome used. */
+  HostUserAgent: string;
+  /** The user agent string that Lighthouse used to load the page. Set to the empty string if unknown. */
+  NetworkUserAgent: string;
+  /** Information on detected tech stacks (e.g. JS libraries) used by the page. */
+  Stacks: Artifacts.DetectedStack[];
+  /** Parsed version of the page's Web App Manifest, or null if none found. This moved to a regular artifact in Fraggle Rock. */
+  WebAppManifest: Artifacts.Manifest | null;
+  /** Errors preventing page being installable as PWA. This moved to a regular artifact in Fraggle Rock. */
+  InstallabilityErrors: Artifacts.InstallabilityErrors;
+  /** A set of page-load traces, keyed by passName. */
+  traces: {[passName: string]: LH.Trace};
```

Comment:
these remaining `LH.*` references are all in `externd.d.ts`, which is a real diverse grab bag of types :)